### PR TITLE
Migrate district service authorization from SQL access controls to OpenFGA

### PR DESCRIPTION
**Pull request by richford**
## Summary

This PR migrates the district service authorization from SQL-based access controls (ltree UNION queries) to OpenFGA, following the pattern established in the administration service migration.

### District service changes

The `list()` method now uses `authorizationService.listAccessibleObjects()` with `FgaRelation.CAN_LIST` and `FgaType.DISTRICT` to resolve accessible district IDs, then fetches records via a new `DistrictRepository.listByIds()` method. The `getById()` method replaces `getAuthorizedById()` with `authorizationService.requirePermission()` using `FgaRelation.CAN_READ`. Super admin bypass and 404-before-403 semantics are preserved.

### Repository changes

Added `listByIds(ids, options)` to `DistrictRepository`. This method wraps the inherited `BaseRepository.getByIds()` with district-specific filtering (orgType, rosteringEnded) and optional embed counts. It serves the same role `listAuthorized` did. FGA resolves the IDs, then this method fetches the actual records with pagination and sorting.

### Authorization behavior change

The FGA model defines `can_list` on district as `supervisory_tier_group`, which includes admin and educator tiers but **not** caregivers or students. Under the old SQL access controls, caregivers with a `guardian` role on a district could see it in list results. With FGA, they cannot. The integration tests have been updated to reflect this. This is consistent with the FGA authorization model design where district-level list visibility is restricted to supervisory roles.

### Test changes

- **Unit tests**: Rewrote district service tests to mock `AuthorizationService` (`listAccessibleObjects`, `requirePermission`) instead of `listAuthorized`/`getAuthorizedById`
- **Repository integration tests**: Added 10 tests for `listByIds` covering ID filtering, pagination, sorting, rostering exclusion, embed counts, and edge cases
- **Route integration tests**: Added `syncFgaTuplesFromPostgres()` to `beforeAll` setup; updated caregiver tier expectation from scoped results to empty results

Resolves https://github.com/yeatmanlab/roar-project-management/issues/1716

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [x] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [x] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [x] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [x] I have linked relevant issues (if any)

### CODE REVIEW

**Comment by richford** [comment]
@claude, please review this PR

**Comment by richford** [comment]
@claude, the behavior change is intentional. Please re-review.

File: apps/backend/src/repositories/district.repository.ts [hunk 1]
```diff
@@ -81,6 +81,26 @@ export class DistrictRepository extends BaseRepository<District, typeof orgs> {
...
+   */
+  private buildDistrictWhereClause(includeEnded: boolean): SQL | undefined {
+    const conditions: SQL[] = [eq(orgs.orgType, OrgType.DISTRICT)];
+
+    if (!includeEnded) {
+      conditions.push(isNull(orgs.rosteringEnded));
+    }
+
+    return conditions.length > 1 ? and(...conditions) : conditions[0];
+  }
```
Comment (ksmontville) [inline apps/backend/src/repositories/district.repository.ts:100]:
```suggestion
  private buildDistrictWhereClause(includeEnded: boolean): SQL | undefined {
    if (includeEnded) {
      return eq(orgs.orgType, OrgType.DISTRICT);
    }

    return and(eq(orgs.orgType, OrgType.DISTRICT), isNull(orgs.rosteringEnded));
  }
```

**Review by ksmontville — CHANGES_REQUESTED** [review]

**Review by ksmontville — APPROVED** [review]

File: apps/backend/src/services/district/district.service.ts
Comment (Emily-ejag) [inline apps/backend/src/services/district/district.service.ts:207]: This method method still uses SQL-based access control (`districtRepository.getAuthorizedById()`) instead of FGA, which contradicts the stated migration goal. Unless I am missing something, I think we should use something like: 
`await authorizationService.requirePermission(userId, FgaRelation.CAN_READ, `${FgaType.DISTRICT}:${districtId}`);` instead.

Comment (Emily-ejag) [inline apps/backend/src/services/district/district.service.ts:281]: This method also uses SQL-based access control (`verifyDistrictAccess()`) instead of FGA, and then applies a second SQL-based check (`getUserRolesForDistrict()`) for supervisory roles. This is inconsistent with the FGA-first pattern. Maybe we should have something like: 

```
// Should verify via FGA first
await verifyDistrictAccess(authContext, districtId);
 
// Then check supervisory permission via FGA
if (!isSuperAdmin) {
  await authorizationService.requirePermission(
    userId,
    FgaRelation.CAN_LIST_USERS,
    `${FgaType.DISTRICT}:${districtId}`,
  );
}
 
// Then use FGA to get accessible schools
const accessibleObjects = await authorizationService.listAccessibleObjects(userId, FgaRelation.CAN_LIST, FgaType.SCHOOL);
const filterIds = accessibleObjects.map(extractFgaObjectId);
return await schoolRepository.listByIds(filterIds, districtId, queryParams);
```

Comment (Emily-ejag) [inline apps/backend/src/services/district/district.service.ts:373]: same as the comments above, The method still uses SQL-based access control (`getAuthorizedUsersByDistrictId()`) for non-super-admin users instead of FGA.

**Review by Emily-ejag — CHANGES_REQUESTED** [review]

File: apps/backend/src/services/district/district.service.ts
Comment (ksmontville) [reply apps/backend/src/services/district/district.service.ts:207]: Thanks for pointing this out. I think that these functions should be addressed at the end of the PR chain, since the school service FGA migration introduces some functions which will be very useful in re-writing these district service functions which still use SQL based authorization.

Comment (ksmontville) [reply apps/backend/src/services/district/district.service.ts:281]: Thanks for pointing this out. I think that these functions should be addressed at the end of the PR chain, since the school service FGA migration introduces some functions which will be very useful in re-writing these district service functions which still use SQL based authorization.

Comment (ksmontville) [reply apps/backend/src/services/district/district.service.ts:373]: Thanks for pointing this out. I think that these functions should be addressed at the end of the PR chain, since the school service FGA migration introduces some functions which will be very useful in re-writing these district service functions which still use SQL based authorization.

**Review by Emily-ejag — APPROVED** [review]
